Agents/MetOfficeAgent/metoffice/kgutils/stackclients.py

```diff
@@ -0,0 +1,258 @@
+
+import glob
+import jaydebeapi
+import json
+
+
+#import agentlogging
+from metoffice.errorhandling.exceptions import StackException
+from metoffice.kgutils.javagateway import stackClientsGw, jpsBaseLibGW
+from metoffice.utils.stack_configs import DB_URL, DB_USER, DB_PASSWORD, ONTOP_URL
+from metoffice.utils.env_configs import DATABASE, LAYERNAME, GEOSERVER_WORKSPACE, ONTOP_FILE
+
+# Initialise logger
+#logger = agentlogging.get_logger("prod")
+
+
+class OntopClient:
+
+    def __init__(self, query_endpoint=ONTOP_URL):
+
+        # create a JVM module view and use it to import the required java classes
+        self.jpsBaseLib_view = jpsBaseLibGW.createModuleView()
+        jpsBaseLibGW.importPackages(self.jpsBaseLib_view,"uk.ac.cam.cares.jps.base.query.*")
+
+        try:
+            # Initialise OntopClient as RemoteStoreClient
+            self.ontop_client = self.jpsBaseLib_view.RemoteStoreClient(query_endpoint)
+        except:
+            #logger.error("Unable to initialise OntopClient.")
+            raise StackException("Unable to initialise OntopClient.")
+    
+    def performQuery(self, query):
+        """
+            This function performs query to Ontop endpoint.
+            Arguments:
+                query - SPARQL Query string
+        """
+        try:
+            response = self.ontop_client.execute(query)
+        except:
+            #logger.error("SPARQL query not successful")
+            raise StackException("SPARQL query not successful")
+        return json.loads(response)
+
+    
+    @staticmethod
+    def upload_ontop_mapping():
+        # Initialise ONTOP client and upload mapping file using default properties
+        # from environment variables
+
+        # Create a JVM module view and use it to import the required java classes
+        stackClientsView = stackClientsGw.createModuleView()
+        stackClientsGw.importPackages(stackClientsView, "com.cmclinnovations.stack.clients.ontop.OntopClient")
+
+        try:
+            # Create JAVA path object to mapping file
+            f = stackClientsView.java.io.File(ONTOP_FILE)
+            fp = f.toPath()
+            # Update ONTOP mapping (requires JAVA path object)
+            stackClientsView.OntopClient().updateOBDA(fp)
+        except Exception as e:
+            #logger.error("Unable to update OBDA mapping.")
+            raise StackException("Unable to update OBDA mapping.", e)
+
+
+class PostGISClient:
+    
+    def __init__(self, dburl=DB_URL, dbuser=DB_USER, dbpassword=DB_PASSWORD,
+                 database=DATABASE):
+        self.dburl = dburl
+        self.dbuser = dbuser
+        self.dbpassword = dbpassword
+        self.database = database
+        try:
+            # Retrieve JDBC connection properties
+            self.conn_props = self.connection_properties()
+        except:
+            #logger.error("Unable to retrieve JDBC connection properties.")
+            raise StackException("Unable to retrieve JDBC connection properties.")
```

Comment:
The handling of exceptions should probably be standardised, can we discuss the various options?

Reply:
This should be addressed by commit `eda2d98e83dfe6a827a9d7ffa38046a2ff717817`